frontend/src/k8sTypes.ts

```diff
@@ -1279,6 +1279,25 @@ export type K8sResourceListResult<TResource extends Partial<K8sResourceCommon>>
   };
 };
 
+/** Represents the status of a component in the DataScienceCluster. */
+export type DataScienceClusterComponentKindStatus = {
+  /**
+   * The management state of the component (e.g., Managed, Unmanaged, Force).
+   * Indicates whether the component is being actively managed or not.
+   */
+  managementState?: string;
+
+  /**
+   * List of releases for the component.
+   * Each release includes the name, the URL of the repository, and the version number.
+   */
+  releases?: Array<{
+    name: string; // Name of the release (e.g., "Kubeflow Pipelines")
+    repoUrl: string; // URL of the repository hosting the release (e.g., GitHub URL)
+    version: string; // Version of the release (e.g., "2.2.0")
```

Comment:
Can you confirm that these are always going to be present and therefore do not need to be marked as optional?

Reply:
Well, those must be there, but it's probably better we mark them as optional.